weedcoco/repo/deposit.py

```diff
@@ -88,15 +93,34 @@ def deposit_weedcoco(weedcoco_path, dataset_dir, image_dir, image_hash):
     return new_dataset_dir
 
 
-def deposit(weedcoco_path, image_dir, repository_dir):
+def retrieve_image_paths(images_path):
+    for root, directories, files in os.walk(images_path):
+        for filename in files:
+            yield (os.path.join(root, filename), filename)
+
+
+def compress_to_download(dataset_dir, deposit_id, download_dir):
+    if not os.path.isdir(download_dir):
+        os.mkdir(download_dir)
+    download_path = str(download_dir / deposit_id) + ".zip"
+    if os.path.exists(download_path):
+        os.remove(download_path)
+    with ZipFile(download_path, "w") as zip:
+        zip.write(dataset_dir / "weedcoco.json", "weedcoco.json")
+        for image, image_name in retrieve_image_paths(dataset_dir / "images"):
+            zip.write(image, "images/" + image_name)
```

Comment:
Avoiding race conditions here would be best done by using a temporary file. Unfortunately, to use a named temporary file properly, it's best to put it in a temporary directory too:
```suggestion
    with tempfile.TemporaryDirectory() as tmpdir:
        zip_path = Path(tmpdir.name) / "bundle.zip"
	    with ZipFile(zip_path, "w") as zip:
	        zip.write(dataset_dir / "weedcoco.json", "weedcoco.json")
	        for image, image_name in retrieve_image_paths(dataset_dir / "images"):
	            zip.write(image, "images/" + image_name)
	    shutil.move(zip_path, download_path)  # atomically overwrite previous file
```